website/static/js/comment.js

```diff
@@ -147,19 +147,20 @@ BaseComment.prototype.fetchNext = function(url, comments) {
         {'isCors': true});
     request.done(function(response) {
         comments = comments.concat(response.data);
-        if (response.links.next !== null) {
-            self.fetchNext(response.links.next, comments);
-        } else {
-            self.comments(
-                ko.utils.arrayMap(comments, function(comment) {
-                    return new CommentModel(comment, self, self.$root);
-                })
-            );
-            deferred.resolve(self.comments());
-            self.configureCommentsVisibility();
+        if (self._loaded !== true) {
             self._loaded = true;
             self.loadingComments(false);
         }
+        self.comments(
+            ko.utils.arrayMap(comments, function(comment) {
+                return new CommentModel(comment, self, self.$root);
+            })
+        );
+        deferred.resolve(self.comments());
```

Comment:
This promise shouldn't get resolved until all the requests have finished, i.e. when there is no more next link.